turbinia/workers/analysis/jupyter.py

```diff
@@ -0,0 +1,120 @@
+
+from __future__ import unicode_literals
+
+import os
+import re
+
+from turbinia import TurbiniaException
+from turbinia.evidence import ReportText
+from turbinia.lib import text_formatter as fmt
+from turbinia.workers import TurbiniaTask
+from turbinia.workers import Priority
+from turbinia.lib.utils import extract_files
+from turbinia.lib.utils import bruteforce_password_hashes
+
+
+class JupyterAnalysisTask(TurbiniaTask):
+  """Task to analyze a Jupyter Notebook config."""
+
+  def run(self, evidence, result):
+    """Run the Jupyter worker.
+
+    Args:
+        evidence (Evidence object):  The evidence to process
+        result (TurbiniaTaskResult): The object to place task results into.
+
+    Returns:
+        TurbiniaTaskResult object.
+    """
+
+    # Where to store the resulting output file.
+    output_file_name = 'jupyter_analysis.txt'
+    output_file_path = os.path.join(self.output_dir, output_file_name)
+
+    # What type of evidence we should output.
+    output_evidence = ReportText(source_path=output_file_path)
+
+    # Read the config file.
+
+    jupyter_config = open(evidence.local_path, 'r').read()
+
+    # Extract the config and return the report
+    (report, priority, summary) = self.analyse_config(jupyter_config)
+    output_evidence.text_data = report
+    result.report_priority = priority
+    result.report_data = report
+
+    # Write the report to the output file.
+    with open(output_file_path, 'w') as fh:
+      fh.write(output_evidence.text_data.encode('utf8'))
+      fh.write('\n'.encode('utf8'))
+
+    # Add the resulting evidence to the result object.
+    result.add_evidence(output_evidence, evidence.config)
+    result.close(self, success=True, status=summary)
+
+    return result
+
+  def analyse_config(self, jupyter_config):
+    """Extract security related configs from Jupyter configuration files.
+
+    Args:
+      config (str): configuration file content.
+
+    Returns:
+      Tuple(
+        report_text(str): The report data
+        report_priority(int): The priority of the report (0 - 100)
+        summary(str): A summary of the report (used for task status)
+      )
+    """
+    findings = []
+    num_misconfigs = 0
+    for line in jupyter_config.split('\n'):
+
+      if all(x in line for x in ['disable_check_xsrf', 'True']):
+        findings.append(fmt.bullet('XSRF protection is disabled.'))
+        num_misconfigs += 1
+        continue
+      if all(x in line for x in ['allow_root', 'True']):
+        findings.append(fmt.bullet('Juypter Notebook allowed to run as root.'))
+        num_misconfigs += 1
+        continue
+      if 'password' in line:
+        if all(x in line for x in ['required', 'False']):
+          findings.append(
+              fmt.bullet(
+                  'Password is not required to access this Jupyter Notebook.'))
+          num_misconfigs += 1
+          continue
+        if 'required' not in line:
+          password_hash = line.split('=')[1].replace(' ', '')
```

Comment:
Sorry I didn't mention this instance of this before, but this is kind of similar to the type of risky parsing I mentioned in the port example.  This will break/stacktrace on other things that contain just 'password' in the line, even something like `#TODO: update password`.   You might consider using a regex or validate the output from `split()` before indexing it.  Also if you're only concerned with leading/trailing spaces, you can just use `.strip()` instead of replacing spaces.

Reply:
updated the if statement. It'll only catch the config related password settings